pkg/webhook/machine/validation/validation.go

```diff
@@ -19,33 +19,46 @@ package validation
 import (
 	"context"
 	"errors"
+	"fmt"
 
 	"go.uber.org/zap"
 
 	clusterv1alpha1 "github.com/kubermatic/machine-controller/pkg/apis/cluster/v1alpha1"
+	kubermaticv1 "k8c.io/kubermatic/v2/pkg/apis/kubermatic/v1"
 	"k8c.io/kubermatic/v2/pkg/resources/certificates"
 
+	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/selection"
 	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
 )
 
 // validator for validating Kubermatic Machine CRD.
 type validator struct {
-	log        *zap.SugaredLogger
-	seedClient ctrlruntimeclient.Client
-	userClient ctrlruntimeclient.Client
-	caBundle   *certificates.CABundle
+	log                 *zap.SugaredLogger
+	seedClient          ctrlruntimeclient.Client
+	userClient          ctrlruntimeclient.Client
+	caBundle            *certificates.CABundle
+	subjectNameSelector labels.Selector
 }
 
 // NewValidator returns a new Machine validator.
-func NewValidator(seedClient, userClient ctrlruntimeclient.Client, log *zap.SugaredLogger, caBundle *certificates.CABundle) *validator {
-	return &validator{
-		log:        log,
-		seedClient: seedClient,
-		userClient: userClient,
-		caBundle:   caBundle,
+func NewValidator(seedClient, userClient ctrlruntimeclient.Client, log *zap.SugaredLogger, caBundle *certificates.CABundle,
+	projectID string) (*validator, error) {
+	subjectNameReq, err := labels.NewRequirement(kubermaticv1.ResourceQuotaSubjectNameLabelKey, selection.Equals, []string{projectID})
```

Comment:
nit: I wonder if you shouldn't check subject's kind as well. I know that for now we only support projects but it might be easier to find this place in the future, as we extend functionality with new kinds, and possibly avoid problems in case someone create RQ of a different kind with the same subject name (shouldn't be possible but better safe then sorry).